gui/app/ports.js

```diff
@@ -10,28 +9,26 @@ const pkg = remote.require('./package.json')
 const defaultDir = path.join(path.homedir(), 'Cozy')
 const container = document.getElementById('container')
 
-const elmectron = Elm.embed(Elm.Main, container, {
-  autolaunch: false,
-  diskSpace: { used: 0, usedUnit: '', total: 0, totalUnit: '' },
-  folder: '',
-  gototab: '',
-  mail: '',
-  pong: null,
-  registration: null,
-  remove: { filename: '', icon: '', path: '', size: 0, updated: 0 },
-  synchonization: '',
-  syncError: '',
-  transfer: { filename: '', icon: '', path: '', size: 0, updated: 0 },
-  updated: [],
-  version: pkg.version
-})
-
-const errMessage = (err) => (err && err.message) ? err.message : err
+const Elm = require('./elm').Main
+const elmectron = Elm.embed(container)
+
+const errMessage = (err) => {
+  if (!err) {
+    return null
+  } else if (err.code === 'ENOTFOUND') {
+    return "The host can't be found"
+  } else if (err.message) {
+    return err.message
+  } else {
+    return `${err}`
+  }
+}
 
 const init = () => {
   elmectron.ports.folder.send(defaultDir)
+  elmectron.ports.version.send(pkg.version)
 }
-init()
+setTimeout(init, 10)
```

Comment:
It doesn't change anything. Elm is loaded asynchronously (I think with a `setimediate`).